### PR TITLE
Move api-reference doc files to /docs directory

### PR DESCRIPTION
**Pull request by Kristian-ZH**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
Move api-reference doc files to /docs directory and attach it to the website.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Markdown files under `/hack/api-reference` are moved to /docs/api-reference directory.
```

### CODE REVIEW

File: .docforge/manifest.yaml [hunk 1]
```diff
@@ -3,67 +3,75 @@
 structure:
 - name: _index.md
   source: https://github.com/gardener/gardener/blob/{{$defaultBranch}}/README.md
+- name: api-reference
+  properties:
+    frontmatter:
+      title: Api Reference
```
Comment (rfranzke) [inline .docforge/manifest.yaml]:
```suggestion
      title: API Reference
```

File: hack/clean.sh [hunk 2]
```diff
@@ -29,6 +29,6 @@ for source_tree in $@; do
   grep -lr '// Code generated by lister-gen. DO NOT EDIT' "$(dirname "$source_tree")" | xargs rm -f
 done
 
-if [ -d "$PWD/hack/api-reference" ]; then
-  find ./hack/api-reference/ -type f -name "*.md" -exec rm '{}' \;
+if [ -d "$PWD/docs/api-reference" ]; then
+  find ./docs/api-reference/ -type f -name "*.md" -exec rm '{}' \;
```
Comment (rfranzke) [inline hack/clean.sh:33]: This change will also be relevant for extensions when revendoring this PR. Can you add a respective release note for developers, please?

**Review by rfranzke — APPROVED** [review]
/lgtm

**Comment by krgostev** [comment]
reviewed/ok-to-test

**Comment by timebertt** [comment]
An index entry in `docs/README.md` would be nice, don't you think? :)

**Comment by Kristian-ZH** [comment]
Won't this be duplicated with the left side navigation bar?

**Comment by timebertt** [comment]
Which "left side navigation bar"? 😄

**Comment by Kristian-ZH** [comment]
Screenshot 2021-11-10 at 9 28 32

**Comment by Kristian-ZH** [comment]
But also what I see is that we use as _index.md (Readme.md) the root Gardener Index.md document so I am not sure where exactly and how we can plugin the documentation Readme.md and to not change the structure.

In my opinion if we add it, we will have to create another mid subdir called `Docs` and it will be just after you click on `Gardener` dir and this will make `gardener` component different from other components

**Pull request merged**